var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -723,23 +757,57 @@ def setup_build_environment(self, env):
 
     def setup_run_environment(self, env):
         self.setup_env_vars(env)
-        env.prepend_path(
-            "PATH", join_path(self.prefix.share, "chapel", self._output_version_short, "util")
+        chpl_home = join_path(self.prefix.share, "chapel", self._output_version_short)
+        env.prepend_path("PATH", join_path(chpl_home, "util"))
+        env.set(
+            "CHPL_MAKE_THIRD_PARTY",
+            join_path(self.prefix.lib, "chapel", self._output_version_short),
+        )
+        env.set("CHPL_HOME", chpl_home)
+
+    def get_main_version_from_cmakelists(self) -> str:
+        with open(join_path(self.build_directory, "CMakeLists.txt"), "r") as f:
+            # read CMakeLists.txt to get the CHPL_MAJOR_VERSION and CHPL_MINOR_VERSION
+            # and then construct the path from that
+            chpl_major_version = None
+            chpl_minor_version = None
+            chpl_patch_version = None
+            for line in f:
+                if "set(CHPL_MAJOR_VERSION" in line:
+                    chpl_major_version = line.split()[1].strip(")")
+                if "set(CHPL_MINOR_VERSION" in line:
+                    chpl_minor_version = line.split()[1].strip(")")
+                if "set(CHPL_PATCH_VERSION" in line:
+                    chpl_patch_version = line.split()[1].strip(")")
+                if (
+                    chpl_major_version is not None
+                    and chpl_minor_version is not None
+                    and chpl_patch_version is not None
+                ):
+                    break
+        assert (
+            chpl_major_version is not None
+            and chpl_minor_version is not None
+            and chpl_patch_version is not None
+        )
+        chpl_version_string = "{}.{}.{}".format(
+            chpl_major_version, chpl_minor_version, chpl_patch_version
         )
+        return chpl_version_string
 
     @property
     @llnl.util.lang.memoized
-    def _output_version_long(self):
+    def _output_version_long(self) -> str:
         if str(self.spec.version).lower() == "main":
```

Comment:
Weak request to generalize the comparison against `"main"` to instead look for any non-numeric `spec.version` and run the new version-detection code.
 
That should ensure things continue to work as expected if someone adds a new named branch at the top of the file.